<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MyDialog</class>
 <widget class="QDialog" name="MyDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>440</width>
    <height>445</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>440</width>
    <height>445</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>440</width>
    <height>445</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="1" column="0">
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QTabWidget" name="tabWidget">
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="widget">
        <attribute name="title">
         <string>诊断</string>
        </attribute>
        <widget class="QWidget" name="verticalLayoutWidget_4">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>391</width>
           <height>331</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <property name="leftMargin">
           <number>10</number>
          </property>
          <property name="topMargin">
           <number>10</number>
          </property>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <property name="topMargin">
             <number>10</number>
            </property>
            <item>
             <widget class="QCheckBox" name="intersect_checkBox">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="maximumSize">
               <size>
                <width>191</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="mouseTracking">
               <bool>true</bool>
              </property>
              <property name="acceptDrops">
               <bool>false</bool>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="checkable">
               <bool>true</bool>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
              <property name="autoRepeat">
               <bool>false</bool>
              </property>
              <property name="tristate">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="intersect_label">
              <property name="text">
               <string>0</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>交叉三角面片</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_2">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="pushButton_8">
              <property name="text">
               <string>标记交叉三角面</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pushButton_15">
              <property name="text">
               <string>擦除标记</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <property name="topMargin">
             <number>10</number>
            </property>
            <item>
             <widget class="QCheckBox" name="overlap_checkBox">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="maximumSize">
               <size>
                <width>191</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="overlap_label">
              <property name="text">
               <string>0</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>重叠三角面片</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_5">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="pushButton_12">
              <property name="text">
               <string>标记重叠三角面</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_9">
            <property name="topMargin">
             <number>10</number>
            </property>
            <item>
             <widget class="QCheckBox" name="reverse_checkBox">
              <property name="maximumSize">
               <size>
                <width>191</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="reverse_label">
              <property name="text">
               <string>0</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>反向三角面片</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_6">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="pushButton_13">
              <property name="text">
               <string>标记反向三角面</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_10">
            <property name="topMargin">
             <number>10</number>
            </property>
            <item>
             <widget class="QCheckBox" name="badside_checkBox">
              <property name="maximumSize">
               <size>
                <width>191</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="badside_label">
              <property name="text">
               <string>0</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_12">
              <property name="text">
               <string>坏边</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_7">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="pushButton_14">
              <property name="text">
               <string>标记坏边</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <property name="topMargin">
             <number>10</number>
            </property>
            <item>
             <widget class="QCheckBox" name="hole_checkBox">
              <property name="maximumSize">
               <size>
                <width>191</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="hole_label">
              <property name="text">
               <string>0</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>孔洞</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_4">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="pushButton_18">
              <property name="text">
               <string>修复孔洞</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <property name="topMargin">
             <number>10</number>
            </property>
            <item>
             <widget class="QCheckBox" name="shell_checkBox">
              <property name="maximumSize">
               <size>
                <width>191</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="shell_label">
              <property name="text">
               <string>0</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>壳体</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_3">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="pushButton_16">
              <property name="text">
               <string>标记壳体</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="verticalSpacer_4">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <property name="topMargin">
             <number>10</number>
            </property>
            <item>
             <spacer name="horizontalSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="pushButton_17">
              <property name="text">
               <string>自动修复</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pushButton">
              <property name="text">
               <string>关闭</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pushButton_2">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="styleSheet">
               <string notr="true">border-image: url(:/Meshprint/Resources/images/button_update_1.png);
border-image: url(:/Meshprint/Resources/images/button_update_2.png);</string>
              </property>
              <property name="text">
               <string>更新</string>
              </property>
              <property name="icon">
               <iconset resource="meshprint.qrc">
                <normaloff>:/Meshprint/Resources/images/button_update_1.png</normaloff>:/Meshprint/Resources/images/button_update_1.png</iconset>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QLabel" name="label">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>380</y>
           <width>411</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>为了避免分析不必要的项目，请选择需要分析的类别，然后点击更新按钮。</string>
         </property>
        </widget>
       </widget>
       <widget class="QWidget" name="widget_2">
        <attribute name="title">
         <string>修复</string>
        </attribute>
        <widget class="QWidget" name="horizontalLayoutWidget">
         <property name="geometry">
          <rect>
           <x>-1</x>
           <y>-1</y>
           <width>401</width>
           <height>271</height>
          </rect>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <property name="leftMargin">
             <number>0</number>
            </property>
            <item>
             <spacer name="verticalSpacer_2">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="pushButton_9">
              <property name="text">
               <string>自动修复</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="verticalSpacer_3">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="pushButton_6">
              <property name="text">
               <string>选择三角面片</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pushButton_5">
              <property name="text">
               <string>创建三角面片</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <property name="leftMargin">
             <number>0</number>
            </property>
            <item>
             <spacer name="verticalSpacer_6">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_7">
              <item>
               <widget class="QPushButton" name="pushButton_4">
                <property name="text">
                 <string>color</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <spacer name="verticalSpacer_5">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_8">
              <item>
               <widget class="QPushButton" name="pushButton_11">
                <property name="text">
                 <string>帮助</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_10">
                <property name="text">
                 <string>关闭</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </widget>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="meshprint.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>pushButton_10</sender>
   <signal>clicked()</signal>
   <receiver>MyDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>408</x>
     <y>297</y>
    </hint>
    <hint type="destinationlabel">
     <x>412</x>
     <y>-5</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MyDialog</sender>
   <signal>rejected()</signal>
   <receiver>MyDialog</receiver>
   <slot>DialogClose()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>211</x>
     <y>150</y>
    </hint>
    <hint type="destinationlabel">
     <x>211</x>
     <y>150</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>MyDialog</receiver>
   <slot>on_updateButton_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>400</x>
     <y>359</y>
    </hint>
    <hint type="destinationlabel">
     <x>437</x>
     <y>311</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MyDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>319</x>
     <y>358</y>
    </hint>
    <hint type="destinationlabel">
     <x>469</x>
     <y>200</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_9</sender>
   <signal>clicked()</signal>
   <receiver>MyDialog</receiver>
   <slot>on_maintenanceButton_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>124</x>
     <y>147</y>
    </hint>
    <hint type="destinationlabel">
     <x>203</x>
     <y>-6</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_5</sender>
   <signal>clicked()</signal>
   <receiver>MyDialog</receiver>
   <slot>on_createTriangle_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>124</x>
     <y>298</y>
    </hint>
    <hint type="destinationlabel">
     <x>-92</x>
     <y>326</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_18</sender>
   <signal>clicked()</signal>
   <receiver>MyDialog</receiver>
   <slot>on_repairhole_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>380</x>
     <y>239</y>
    </hint>
    <hint type="destinationlabel">
     <x>392</x>
     <y>-12</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_8</sender>
   <signal>clicked()</signal>
   <receiver>MyDialog</receiver>
   <slot>on_markcolorbutton_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>290</x>
     <y>86</y>
    </hint>
    <hint type="destinationlabel">
     <x>485</x>
     <y>69</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_15</sender>
   <signal>clicked()</signal>
   <receiver>MyDialog</receiver>
   <slot>on_clearButton_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>354</x>
     <y>82</y>
    </hint>
    <hint type="destinationlabel">
     <x>460</x>
     <y>76</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>intersect_checkBox</sender>
   <signal>clicked()</signal>
   <receiver>MyDialog</receiver>
   <slot>on_intersectAndoverlapCheckBox()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>30</x>
     <y>84</y>
    </hint>
    <hint type="destinationlabel">
     <x>-24</x>
     <y>65</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_14</sender>
   <signal>clicked()</signal>
   <receiver>MyDialog</receiver>
   <slot>on_markbadisdebutton_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>362</x>
     <y>203</y>
    </hint>
    <hint type="destinationlabel">
     <x>482</x>
     <y>154</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_17</sender>
   <signal>clicked()</signal>
   <receiver>MyDialog</receiver>
   <slot>on_maintenanceButton_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>194</x>
     <y>351</y>
    </hint>
    <hint type="destinationlabel">
     <x>513</x>
     <y>341</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>DialogClose()</slot>
  <slot>on_updateButton_clicked()</slot>
  <slot>on_clearButton_clicked()</slot>
  <slot>on_maintenanceButton_clicked()</slot>
  <slot>on_markcolorbutton_clicked()</slot>
  <slot>on_createTriangle_clicked()</slot>
  <slot>on_repairhole_clicked()</slot>
  <slot>on_holeMark_clicked()</slot>
  <slot>on_intersectAndoverlapCheckBox()</slot>
  <slot>on_markbadisdebutton_clicked()</slot>
 </slots>
</ui>
